pkg/scheduler/podtask/ranges.go

```diff
@@ -0,0 +1,140 @@
+package podtask
+
+import (
+	"sort"
+
+	"github.com/gogo/protobuf/proto"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+)
+
+// Range is an all-inclusive range defined by a lower and upper bound.
+type Range [2]uint64
+
+// Ranges represents a list of Ranges.
+type Ranges []Range
+
+// NewPortRanges returns Ranges from the "ports" resource in the
+// given *mesos.Offer. If that resource isn't provided, nil will be returned.
+//
+// The returned Ranges are sorted and have all overlapping ranges merged from
+// left to right. e.g. [[0, 5], [4, 3], [10, 7]] -> [[0, 5], [7, 10]]
```

Comment:
This seems like something I would expect to have already occurred.  Should this logic instead exist in our go mesos library and happen by default?